src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -1210,6 +1232,9 @@ private static uint HandleEventPeerSendShutdown(State state)
         private static uint HandleEventSendComplete(State state, ref StreamEvent evt)
         {
             StreamEventDataSendComplete sendCompleteEvent = evt.Data.SendComplete;
+            // CR: canceled will be set when connection gets aborted, 
+            // send with FIN followed in a quick succession by another send
+            // CR: we might ignore canceled, since there might be a race between connection getting closed and geting this event.
```

Comment:
https://github.com/dotnet/runtime/issues/59346
I'm not 100% sure I've got this right and correctly described it in the issue. @CarnaViire it's 14th bullet point in the issue. could you please check it there?